code/game/objects/items/devices/lightreplacer.dm

```diff
@@ -114,72 +156,102 @@
 			usr << "You shortcircuit the [src]."
 			return
 	*/
-	usr << "It has [uses] lights remaining."
 
-/obj/item/device/lightreplacer/update_icon()
-	icon_state = "lightreplacer[emagged]"
+	var/dat = {"<TITLE>Light Replacer Interface</TITLE>
 
+	Glass storage: [glass_stor]/[glass_stor_max]<br>"}
 
-/obj/item/device/lightreplacer/proc/Use(var/mob/user)
+	if(supply)
+		dat += {"<a href='?src=\ref[src];build=tube'>Fabricate Tube</a>
+		<a href='?src=\ref[src];build=bulb'>Fabricate Bulb</a>
 
-	playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
-	AddUses(-1)
-	return 1
+		<h3>Supply Container:</h3>"} //It's not clear here, but the argument to build is the part of the typepath after /obj/item/weapon/light/
+		var/list/light_types = new()
+		for(var/obj/item/weapon/light/L in supply)
+			if(!light_types[L.name])
+				light_types[L.name] = 0
+			light_types[L.name]++
+	
+		for(var/T in light_types)
+			dat += "<br><b>[T]: </b>[light_types[T]]"
+
+		dat += "<br><b><a href='?src=\ref[src];eject=supply'>Eject Supply Container</a></b>"
+
+	else
+		dat += "<h3>No supply container inserted</h3>"
+
+	if(waste)
+		dat += {"<br><br><br><h3>Waste Container:</h3>
+	
+		<b>Filled: </b>[waste.contents.len]/[waste.storage_slots]<br>
+		<b><a href='?src=\ref[src];eject=waste'>Eject Waste Container</a></b>
+		"}
+	else
+		dat += "<br><br><br><h3>No waste container inserted</h3>"
+
+	user << browse(dat, "window=lightreplacer")

```

Comment:
https://github.com/d3athrow/vgstation13/blob/Bleeding-Edge/code/datums/browser.dm

They're a basic framework for interfaces, and also has some default CSS to make them look not shit.